pyomo/core/base/componentuid.py

```diff
@@ -295,6 +304,127 @@ def _record_indexed_object_cuid_strings_v2(obj, cuid_str):
                     _record_indexed_object_cuid_strings(obj, cuid_str)
         return cuid_strings
 
+    def _index_from_slice_info(self, slice_info):
+        """
+        Constructs an index from the slice_info entry in a slice's
+        call stack. The index may then be processed just as any
+        other slice index, e.g. from a __getitem__ call in a slice's
+        call stack.
+        """
+        fixed, sliced, ellipsis = slice_info
+
+        if ellipsis is None:
+            ellipsis = {}
+        else:
+            ellipsis = {ellipsis: Ellipsis}
+
+        value_map = {}
+        value_map.update(fixed)
+        try:
+            value_map.update(sliced)
+        except TypeError:
+            sliced = {i: slice(None) for i in sliced}
+            value_map.update(sliced)
```

Comment:
Why is this except clause here?  I am not sure I can come up with a way to trigger this.

Reply:
Slices created by the flattener use a list of sliced indices rather than a dict mapping indices to `slice(None)`.

Reply:
Ok. Would you mind adding a comment to that effect and a test to cover it?

Reply:
Yep, will do.